gateway/src/model.rs

```diff
@@ -262,15 +268,22 @@ impl<'de> Deserialize<'de> for ProviderConfig {
         D: serde::Deserializer<'de>,
     {
         let helper = ProviderConfigHelper::deserialize(deserializer)?;
+        let convert_to_credential = |location: CredentialLocation, provider_name: &str, model_name: &str| {
```

Comment:
if we move the conversion back into the model new() function then we can remove this.